doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -689,198 +734,548 @@ one from Ltac1, except that it requires the goal to be focused.
-    out of the parsed values in the same order. As an optimization, all
-    subscopes of the form :n:`STRING` are left out of the returned tuple, instead
-    of returning a useless unit value. It is forbidden for the various
-    subscopes to refer to the global entry using :n:`self` or :n:`next`.
+  :n:`ident`
+    Parses :token:`ident` or :n:`$@ident`.  The first form returns :n:`ident:(@ident)`,
+    while the latter form returns the variable :n:`@ident`.
 
-A few other specific scopes exist to handle Ltac1-like syntax, but their use is
+  :n:`tactic {? (@int) }`
+    Parses an :token:`ltac2_expr`.  If :token:`int` is specified, the construct
+    parses a :n:`tac2expr@int`, for example `tactic(5)` parses :token:`tac2expr5`.
+    :token:`int` must be in the range `0 .. 6`.
+
+    .. todo this doesn't work as expected: "::" is in tac2expr1
+       Ltac2 Notation "ex4" x(tactic(0)) := x.
+       ex4 auto :: [auto].
+
+  .. not sure "self" and "next" do anything special.  I get the same error
+     message for both from constructs like
+
+     Ltac2 Notation "ex5" x(self) := auto.
+     ex5 match.
+
+     Syntax error: [tactic:tac2expr level 5] expected after 'match' (in [tactic:tac2expr]).
+
+  :n:`self`
+    parses an Ltac2 expression at the current level and returns it as is.
+
+  :n:`next`
+    parses an Ltac2 expression at the next level and returns it as is.
+
+  :n:`thunk(@ltac2_scope)`
+    Used for semantic effect only, parses the same as :token:`ltac2_scope`.
+    If :n:`e` is the parsed expression for :token:`ltac2_scope`, `thunk`
+    returns :n:`fun () => e`.
+
+  :n:`@string`
+    parses the corresponding string as an identifier and returns :n:`()`.
+
+  .. todo PR I dropped "terminal" and "keyword" because they don't seem to behave
+     any differently from just giving a string, e.g. for the keyword "match",
+     x(terminal("match")) worked the same as x("match")
+
+A few syntactic classes contain antiquotation features. For the sake of uniformity, all
+antiquotations are introduced by the syntax :n:`$@lident`.
+
+A few other specific syntactic classes exist to handle Ltac1-like syntax, but their use is
 discouraged and they are thus not documented.
 
-For now there is no way to declare new scopes from Ltac2 side, but this is
+For now there is no way to declare new syntactic classes from the Ltac2 side, but this is
 planned.
 
-Notations
-~~~~~~~~~
+Other nonterminals that have syntactic classes are listed here.
+The class name is the Ltac2 nonterminal name without the prefix :n:`q_`.  (Exception:
+the class name for :token:`q_strategy_flag` is :n:`q_strategy`).
 
-The Ltac2 parser can be extended with syntactic notations.
+   .. list-table::
+      :header-rows: 1
 
-.. cmd:: Ltac2 Notation {+ {| @lident (@ltac2_scope) | @string } } {? : @int} := @ltac2_term
-   :name: Ltac2 Notation
+      * - Ltac2 nonterminal
+        - Equivalent
+        - Notes
```

Comment:
Do you intend to keep this third column?

Reply:
Probably not but won't remove it quite yet.